cli/src/semgrep/commands/wrapper.py

```
@@ -50,6 +50,9 @@ def wrapper(*args: Any, **kwargs: Any) -> NoReturn:
             metrics.add_exit_code(exit_code)
             metrics.send()
 
+            error_handler = get_state().error_handler
```

Will this still fire even if something goes wrong with the metrics handling?

---

Yes, it will always fire. The metrics handler has its own `suppress_errors` function wrapper that silences any errors.